pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -86,11 +92,161 @@ type Interface interface {
+		nodeInfo, err := csiKubeClient.CsiV1alpha1().CSINodeInfos().Get(string(nim.nodeName), metav1.GetOptions{})
+		if err != nil {
+			updateErrs = append(updateErrs, err)
+			return false, nil
+		}
+		err = nim.tryInitializeMigratedDriverStatuses(nodeInfo)
+		if err != nil {
+			updateErrs = append(updateErrs, err)
+			return false, nil
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("error updating node: %v; caused by: %v", err, utilerrors.NewAggregate(updateErrs))
+	}
+	return nil
+}
+
+func (nim *nodeInfoManager) tryInitializeMigratedDriverSpecs(nodeInfo *csiv1alpha1.CSINodeInfo) error {
+	csiKubeClient := nim.volumeHost.GetCSIClient()
+	if csiKubeClient == nil {
+		return fmt.Errorf("error getting CSI client")
+	}
+
+	existingDrivers := sets.String{}
+
+	// Do existing updates
+	// Clone driver list, omitting the driver that matches the given driverName
+	newDriverSpecs := []csiv1alpha1.CSIDriverInfoSpec{}
+	for _, driverInfoSpec := range nodeInfo.Spec.Drivers {
+		// No need to change any specs
+		newDriverSpecs = append(newDriverSpecs, driverInfoSpec)
+		existingDrivers.Insert(driverInfoSpec.Name)
+	}
+
+	for driver, _ := range nim.migratedDrivers {
+		if existingDrivers.Has(driver) {
+			// Skip adding new driver because already updated previously
+			continue
+		}
+
+		driverSpec := csiv1alpha1.CSIDriverInfoSpec{
+			Name:         driver,
+			NodeID:       "",
+			TopologyKeys: []string{},
+		}
+
+		newDriverSpecs = append(newDriverSpecs, driverSpec)
+
+	}
+
+	nodeInfo.Spec.Drivers = newDriverSpecs
+
+	_, err := csiKubeClient.CsiV1alpha1().CSINodeInfos().Update(nodeInfo)
+	return err
+}
+
+func (nim *nodeInfoManager) tryInitializeMigratedDriverStatuses(nodeInfo *csiv1alpha1.CSINodeInfo) error {
+	csiKubeClient := nim.volumeHost.GetCSIClient()
+	if csiKubeClient == nil {
+		return fmt.Errorf("error getting CSI client")
+	}
+
+	existingDrivers := sets.String{}
+
+	newDriverStatuses := []csiv1alpha1.CSIDriverInfoStatus{}
+	// Add existing Statuses to newDriverStatuses
+	for _, driverInfoStatus := range nodeInfo.Status.Drivers {
+		driverInfoStatus.VolumePluginMechanism = nim.getVolumePluginMechanism(driverInfoStatus.Name)
+		existingDrivers.Insert(driverInfoStatus.Name)
+		newDriverStatuses = append(newDriverStatuses, driverInfoStatus)
 	}
+
+	// Add new Statuses to newDriverStatuses for each spec
+	for _, driverInfoSpec := range nodeInfo.Spec.Drivers {
+		if existingDrivers.Has(driverInfoSpec.Name) {
+			// Skip adding new status because already added previously
+			continue
+		}
+
+		volumePluginMechanism := nim.getVolumePluginMechanism(driverInfoSpec.Name)
+		driverStatus := csiv1alpha1.CSIDriverInfoStatus{
+			Name: driverInfoSpec.Name,
+			// We are just adding this driver object it's not installed
+			Available:             false,
+			VolumePluginMechanism: volumePluginMechanism,
+		}
+		newDriverStatuses = append(newDriverStatuses, driverStatus)
```

Comment:
track whether we needed to do this, and if we didn't append new statuses, skip the updatestatus